core/iwasm/libraries/lib-pthread/lib_pthread_wrapper.c

```diff
@@ -479,17 +479,8 @@ pthread_start_routine(void *arg)
     os_cond_signal(&parent_exec_env->wait_cond);
     os_mutex_unlock(&parent_exec_env->wait_lock);
 
-    if (!validate_native_addr(routine_args->arg, sizeof(uint32))) {
-        /* If there are exceptions, copy the exception to
-            all other instance in this cluster */
-        wasm_cluster_spread_exception(exec_env);
-        wasm_runtime_deinstantiate_internal(module_inst, true);
-        delete_thread_info_node(info_node);
-        return NULL;
-    }
-
     wasm_exec_env_set_thread_info(exec_env);
-    argv[0] = addr_native_to_app(routine_args->arg);
+    argv[0] = routine_args->arg;
```

Comment:
The type of routine_args->arg is void *, converting to uint32 causes compilation warning, could you please change the type of field arg in ThreadRoutineArgs to uint32?

And also change the last argument in pthread_create_wrapper from `void *arg` to `uint32 arg`.

Reply:
ok

Reply:
done